src/URI.js

```diff
@@ -1223,6 +1227,26 @@
   };
 
   // compound accessors
+  p.origin = function(v, build) {
+    var parts;
+
+    if (this._parts.urn) {
+      return v === undefined ? '' : this;
+    }
+
+    if (v === undefined) {
+      return this._parts.hostname ? URI.buildOrigin(this._parts) : '';
+    } else {
+      parts = URI.parse(v);
```

Comment:
I think I would have used the existing accessory here as well:

```js
var origin = URI(v);
this
  .protocol(origin.protocol())
  .authority(origin.authority())
  .build(!build);
return this;
```

While simpler to read, it is slower because of double parsing…